src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -84,6 +101,17 @@ public int ExecuteAsync()
             }
         }
 
+        internal void OnTestSessionTimeout(object obj)
+        {
+            string message = String.Format(ClientResources.TestSessionTimeoutMessage, this.testSessionTimeout);
```

Comment:
Good point, need to think through it. Currently in all the places we are doing this way only.